docs/flamework/additional-modules/components/configuration.md

```diff
@@ -0,0 +1,69 @@
+---
+title: Configuration
+---
+Components have a few features we haven't covered yet!
+
+## Predicate
+You can add a `predicate` function to a component to determine if the component should be constructed for a provided instance.
+
+This only affects Components created via **CollectionService**!
+```ts
+@Component({
+    tag: "Example",
+    predicate: ( instance ) => instance.FindFirstAncestorOfClass("PlayerGui") !== undefined,
+})
+export class ExampleComponent extends BaseComponent<> implements OnStart {}
+```
+
+## Instance Guard
+The `instanceGuard` is similar to `predicate`, but serves a slightly different purpose.
+
+* On the server, an error will be thrown if a component is attempted to be created, but the `instanceGuard` fails.
+* On the client, a connection will be created to wait for `DescendantsAdded` & `DescendantsRemoving` until the `instanceGuard` passes. This is because instances can be sent to the client in pieces and the whole thing might not have arrived yet.
+
+Note: Using `instanceGuard` will override the automatically generated type guard, which is created when providing the `BaseComponent` with an Instance type!
+```ts
+@Component( {
+    tag: "Example",
+    instanceGuard: Flamework.createGuard<any>()
+} )
+export class ExampleComponent extends BaseComponent<> implements OnStart {}
```

Comment:
```suggestion
export class ExampleComponent extends BaseComponent<{}, Model> implements OnStart {}
```

Any instance type would work but since we're overriding the type guard we should specify this.